File: 4 занятие/a.py
a = []
def contains(c, n):
    l = 0
    r = n
    while (r - l > 1):
        mid = (r + l) // 2
        if (a[mid] == c):
            return True
        elif a[mid] < c:
            l = mid
        else:
            r = mid
    
    if l >= 0 and a[l] == c or r < n and a[r] == c:
        return True
    return False


n, k = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))

for i in range(len(b)):
    if contains(b[i], len(a)):
        print("YES")
    else:
        print("NO")

File: 4 занятие/a.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

int a[100005];

bool contains(int c, int n) {
    int l = 0;
    int r = n;
    while (r - l > 1) {
        int mid = (r + l) / 2;
        if (a[mid] == c) {
            return true;
        } else if (a[mid] < c) {
            l = mid;
        } else {
            r = mid;
        }
    }
    if (a[l] == c || a[r] == c) {
        return true;
    }
    return false;
}

int main() {
    int n, k;
    cin >> n >> k;
    int b[k];

    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }    

    for (int i = 0; i < k; i++) {
        cin >> b[i];
    }    

    for (int i = 0; i < k; i++) {
        if (contains(b[i], n)) {
            cout << "YES\n";
        } else {
            cout << "NO\n";
        }
    }

}